### PR TITLE
fix scatter range receive npe when get store info of non-eixst store

### PR DESCRIPTION
**Pull request by lerencao**
### What problem does this PR solve? 

Fix a npe when get store info of a non-exist stores in scatter range

### What is changed and how it works?
We got a stack error in our online env. 

```
2019/02/22 15:28:53.887 log.go:274: [fatal] panic: runtime error: invalid memory address or nil pointer dereference, stack: goroutine 6
2180022 [running]:
runtime/debug.Stack(0xc42caa76d8, 0xfe4e40, 0x18cf570)
        /data/program/tistack/tistack-release/local/go/src/runtime/debug/stack.go:24 +0xa7
github.com/pingcap/pd/pkg/logutil.LogPanic()
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/pkg/logutil/log.go:274 +0x5a
panic(0xfe4e40, 0x18cf570)
        /data/program/tistack/tistack-release/local/go/src/runtime/panic.go:505 +0x229
github.com/pingcap/pd/server/schedule.(*RangeCluster).updateStoreInfo(0xc421d0bb20, 0x0)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/schedule/range_cluster.go:62 +0x26
github.com/pingcap/pd/server/schedule.(*RangeCluster).GetStore(0xc421d0bb20, 0x11f4b5, 0xc4277ca010)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/schedule/range_cluster.go:81 +0x5d
github.com/pingcap/pd/server/schedulers.(*balanceLeaderScheduler).transferLeaderIn(0xc42619b060, 0xc4282f7030, 0x12af300, 0xc421d0bb20,
 0xc4282facc0, 0xc4282facf0, 0x0, 0x0, 0x0)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/schedulers/balance_leader.go:139 +0x10b
github.com/pingcap/pd/server/schedulers.(*balanceLeaderScheduler).Schedule(0xc42619b060, 0x12af300, 0xc421d0bb20, 0xc4282facc0, 0xc4282
facf0, 0xc4282facc0, 0xc4282facf0, 0x20)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/schedulers/balance_leader.go:101 +0x5e0
github.com/pingcap/pd/server/schedulers.(*scatterRangeScheduler).Schedule(0xc42a29c850, 0x12af180, 0xc42a3ce700, 0xc42ec8e6f0, 0xc42ec8
e720, 0xa, 0xc42a3ce700, 0x0)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/schedulers/scatter_range.go:101 +0x205
github.com/pingcap/pd/server.scheduleByNamespace(0x12af000, 0xc424294960, 0x12a4380, 0xc421adf040, 0x12a62a0, 0xc42a29c850, 0xc42ec8e6f
0, 0xc42ec8e720, 0xc42ec8e720, 0x11f1b3, ...)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/namespace_cluster.go:142 +0x13e
github.com/pingcap/pd/server.(*scheduleController).Schedule(0xc42ad3ed20, 0x12af000, 0xc424294960, 0xc42ec8e6f0, 0xc42ec8e720, 0xc42ec8
e6f0, 0xc42ec8e720, 0x0)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/coordinator.go:689 +0xa3
github.com/pingcap/pd/server.(*coordinator).runScheduler(0xc42002f680, 0xc42ad3ed20)
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/coordinator.go:433 +0x2be
created by github.com/pingcap/pd/server.(*coordinator).addScheduler
        /data/program/tistack/tistack-release/go/src/github.com/pingcap/pd/server/coordinator.go:394 +0x16a

```

### Check List 

Tests 

 - No code

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by siddontang** [comment]
Thank you @lerencao

**Comment by siddontang** [comment]
@nolouch 

how can we meet this? seem we need to add a test.

**Comment by lerencao** [comment]
after pd server restarts, not all region infos are updated, and region info's default store id is 0.
So NPE can appear when the code get store info of id 0.

**Comment by nolouch** [comment]
@lerencao can you add this situation in the test?

**Comment by lerencao** [comment]
@nolouch Yes, will work on it.

**Comment by lerencao** [comment]
@nolouch  PTAL

Test cases and another nil check are added

**Comment by nolouch** [comment]
@lerencao ci failed, may related to `gofmt`.

**Comment by lerencao** [comment]
@nolouch updated.

File: server/schedulers/balance_test.go [hunk 1]
```diff
@@ -1306,3 +1306,69 @@ func (s *testScatterRangeLeaderSuite) TestBalance(c *C) {
...
+			meta,
+			leader,
+			core.SetApproximateKeys(96),
+			core.SetApproximateSize(96),
+		)
+
+		tc.Regions.SetRegion(regionInfo)
+	}
+
+	for i := 1; i <= 3; i++ {
```
Comment (rleungx) [inline server/schedulers/balance_test.go]: It seems we don't need it?

File: server/schedulers/balance_test.go [hunk 2]
```diff
@@ -1306,3 +1306,69 @@ func (s *testScatterRangeLeaderSuite) TestBalance(c *C) {
...
+		tc.UpdateStoreStatus(uint64(i))
+	}
+
+	oc := schedule.NewOperatorController(nil, nil)
+	hb := newScatterRangeScheduler(oc, []string{"s_00", "s_09", "t"})
+
+	limit := 0
```
Comment (rleungx) [inline server/schedulers/balance_test.go]: ditto. I think testing if `hb.Schedule(tc)` can work is enough?

File: server/schedulers/balance_test.go (hunk 1 above)
Comment (lerencao) [reply server/schedulers/balance_test.go]: Maybe the store info is not used in the underlining implementations,  but it should be here to make things straight.

File: server/schedulers/balance_test.go (hunk 2 above)
Comment (lerencao) [reply server/schedulers/balance_test.go]: Nope. scatter-range just select a region randomly. If we just schedule once, the region without heartbeat may not be scheduled.

**Pull request merged**